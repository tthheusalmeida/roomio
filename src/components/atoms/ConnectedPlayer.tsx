import { mergeClassNames } from "@/utils/classNames";

interface ConnectedPlayerProps {
  name: string;
  className?: string;
}

export default function ConnectedPlayer({
  name,
  className,
}: ConnectedPlayerProps) {
  return (
    <div className={mergeClassNames("flex items-center gap-2", className)}>
      <div className="w-2 h-2 rounded-full border bg-green-500" />
      <span className="font-medium text-xs text-violet-100">{name}</span>
    </div>
  );
}
