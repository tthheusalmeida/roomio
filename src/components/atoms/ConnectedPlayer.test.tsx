import { render, screen } from "@testing-library/react";
import ConnectedPlayer from "./ConnectedPlayer";

describe("ConnectedPlayer component", () => {
  it("renders the player name and indicator", () => {
    const { asFragment } = render(<ConnectedPlayer name="Matheus" />);

    expect(screen.getByText("Matheus")).toBeInTheDocument();

    const indicator = screen.getByText("Matheus").previousSibling;
    expect(indicator).toHaveClass("w-2 h-2 rounded-full border bg-green-500");

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <ConnectedPlayer name="Matheus" className="extra-class" />
    );

    const container = screen.getByText("Matheus").parentElement;
    expect(container).toHaveClass("flex", "items-center", "gap-2");
    expect(container).toHaveClass("extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
