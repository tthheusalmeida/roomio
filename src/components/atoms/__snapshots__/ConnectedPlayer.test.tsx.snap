// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ConnectedPlayer component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-2 extra-class"
  >
    <div
      class="w-2 h-2 rounded-full border bg-green-500"
    />
    <span
      class="font-medium text-xs text-violet-100"
    >
      Matheus
    </span>
  </div>
</DocumentFragment>
`;

exports[`ConnectedPlayer component renders the player name and indicator 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-2"
  >
    <div
      class="w-2 h-2 rounded-full border bg-green-500"
    />
    <span
      class="font-medium text-xs text-violet-100"
    >
      Matheus
    </span>
  </div>
</DocumentFragment>
`;
